mteb/models/instruct_wrapper.py

```diff
@@ -78,3 +80,78 @@ def encode(
             return embeddings
 
     return InstructWrapper(model_name_or_path, mode, instruction_template, **kwargs)
+
+
+class InstructSentenceTransformerWrapper(Wrapper):
+    def __init__(
+        self,
+        model_name: str,
+        revision: str,
+        instruction_template: str | Callable[[str], str] | None = None,
+        max_seq_length: int | None = None,
+        apply_instruction_to_passages: bool = True,
+        **kwargs: Any,
+    ):
+        """Args:
+        model_name: model name
+        revision: revision
+        instruction_template: Model template. Should contain the string '{instruction}'.
+        max_seq_length: maximum sequence length. If None, the maximum sequence length won't be changed.
+        apply_instruction_to_passages: Whether to apply the instruction template to the passages.
+        **kwargs:
+        """
```

Comment:
format is wrong

It would be nice to add an explanation on when to use this as model models with instructions that will not need this wrapper (e.g. e5 models)